--- libdbi-drivers.orig/configure.in	2009-02-27 02:46:35.000000000 +0100
+++ libdbi-drivers/configure.in	2009-04-21 10:32:09.000000000 +0200
@@ -146,7 +146,9 @@
 ac_docs="YES"
 AC_ARG_ENABLE(docs,
 	[  --disable-docs	  do not build and install the documentation.],
-	[  ac_docs="NO" ])
+	[  if test "$enable_docs" = "no"; then
+	       ac_docs="NO"
+	   fi ])
 
 if test "$ac_docs" = "YES"; then
       docs_subdirs="doc"
