--- libdbi-drivers-0.7.0/configure.in.orig	2003-06-22 18:21:48.000000000 +0200
+++ libdbi-drivers-0.7.0/configure.in	2003-07-17 08:11:47.000000000 +0200
@@ -47,25 +47,24 @@
                 CFLAGS="-xO4 -fast -w -fsimple -native -xcg92"
                 PROFILE="-v -xpg -g -xO4 -fast -native -fsimple -xcg92 -Dsuncc" ;;
         *)
-                DEBUG="-g"
-                CFLAGS="-O"
-                PROFILE="-g -p" ;;
+                DEBUG="$CFLAGS -g"
+                PROFILE="$CFLAGS -g -p" ;;
         esac
 else
 
         case $host in
         *-*-linux*)
-                DEBUG="-g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
-                CFLAGS="-O20 -ffast-math -D_REENTRANT -fsigned-char"
-                PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char";;
+                DEBUG="$CFLAGS -g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
+                PROFILE="-pg -g $CFLAGS -ffast-math -D_REENTRANT -fsigned-char"
+                CFLAGS="$CFLAGS -ffast-math -D_REENTRANT -fsigned-char";;
         sparc-sun-*)
-                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8"
-                CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8"
-                PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
+                DEBUG="$CFLAGS -g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8"
+                PROFILE="-pg -g $CFLAGS -D__NO_MATH_INLINES -fsigned-char -mv8"
+                CFLAGS="$CFLAGS -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
         *)
-                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
-                CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char"
-                PROFILE="-O20 -g -pg -D__NO_MATH_INLINES -fsigned-char" ;;
+                DEBUG="$CFLAGS -g -Wall -D__NO_MATH_INLINES -fsigned-char"
+                PROFILE="$CFLAGS -g -pg -D__NO_MATH_INLINES -fsigned-char"
+                CFLAGS="$CFLAGS -D__NO_MATH_INLINES -fsigned-char" ;;
         esac
 fi
 
